packages/java/src/browser/java-client-contribution.ts

```diff
@@ -102,9 +99,23 @@ export class JavaClientContribution extends BaseLanguageClientContribution {
         });
     }
 
+    protected createLanguageClient(connection: MessageConnection): ILanguageClient {
+        const clientConnection = super.createLanguageClient(connection);
+        const client: ILanguageClient & Readonly<{ languageId: string }> = Object.assign(clientConnection, { languageId: this.id });
+        client.registerFeature(SemanticHighlightingService.createNewFeature(this.semanticHighlightingService, client));
+        return client;
+    }
+
+    waitForActivation(app: FrontendApplication): Promise<void> {
+        return Promise.race([super.waitForActivation(app), this.javaService.collectJavaExtensions().then(extensions => {
```

Comment:
`Promise.race` resolves if one of given promises is resolved, you want `Promise.all` or simpler:
```ts
await super.waitForActivation(app);
this.javaExtensions = await this.javaServices.collectJavaExtensions();
```

Reply:
Fixed, thank you